python/ray/train/_internal/storage.py

```diff
@@ -48,46 +50,51 @@ class _ExcludingLocalFilesystem(LocalFileSystem):
     """LocalFileSystem wrapper to exclude files according to patterns.
 
     Args:
+        root_path: Root path to strip when matching with the exclude pattern.
+            Ex: root_path="/tmp/a/b/c", exclude=["*a*"], will exclude
+            /tmp/a/b/c/_a_.txt but not ALL of /tmp/a/*.
```

Comment:
this was bugged from before:

if `name="checkpoint_anything"`, our fnmatch exclude on the full path would exclude the entire experiment directory.

`exclude=["*/checkpoint_*"]` would match `/storage/path/checkpoint_anything/`, rather than `/storage/path/checkpoint_anything/*/checkpoint_*`.

Reply:
Now that I'm thinking about it, this exclude is not really needed since we write the checkpoint directly to storage, and we have no control over what the user names their files in the trial dir. Still may be useful in the future for whitelisting, so let's keep it for now.